src/ensembl/src/content/app/entity-viewer/gene-view/components/transcripts-filter/TranscriptsFilter.tsx

```diff
@@ -63,29 +70,35 @@ const radioData: RadioOptions = sortingOrderToLabel.map(([key, value]) => ({
   label: value
 }));
 
+export const metadataFields = ['biotype', 'appris', 'tsl'] as const;
+
+const createFilter = (metadataType: Filter['type'], label: string) => ({
+  label,
+  selected: false,
+  type: metadataType
+});
+
 const TranscriptsFilter = (props: Props) => {
   const filters = useSelector(getFilters);
   const sortingRule = useSelector(getSortingRule);
   const isSidebarOpen = useSelector(isEntityViewerSidebarOpen);
   const dispatch = useDispatch();
 
-  const biotypes = props.transcripts
-    .map((a) => a.so_term)
-    .filter(Boolean) as string[];
-
-  const uniqueBiotypes = Array.from(new Set(biotypes)).sort((a, b) => {
-    return sortBiotypes(a, b);
-  });
+  const initialFilters: Filters = {};
+
+  for (const transcript of props.transcripts) {
+    metadataFields.forEach((key) => {
+      const metadataItem = transcript.metadata[key];
+      if (metadataItem) {
+        initialFilters[metadataItem.value as string] = createFilter(
+          key,
+          metadataItem.label
+        );
+      }
+    });
+  }
 
   // TODO: Add protein coding options in RadioOptions if there are protein coding biotype
-  const initialFilters = uniqueBiotypes.reduce((accumulator, biotype): {
-    [filter: string]: boolean;
-  } => {
-    return {
-      ...accumulator,
-      [biotype]: false
-    };
-  }, {});
 
   useEffect(() => {
     if (Object.keys(filters).length === 0) {
```

Comment:
It's has now become possible to change the gene id without unmounting the filters component, which suggests that we should at least update the list of dependencies for this useEffect.

![image](https://user-images.githubusercontent.com/6834224/126161570-a5547c6f-a8a7-43dd-9914-66251ad3b109.png)

Reply:
this is now fixed. Thanks @azangru